src/com/cafeto/training/singletonFactory.java

```diff
@@ -0,0 +1,34 @@
+
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Created by agomez on 7/2/2015.
+ */
+public enum singletonFactory{
+
+
+    ACCESO {
+        IDataAccess dataAccess= new IDataAccess() {
```

Comment:
this is not correct, you should not initialize the IDataAccess inside the enum element, you should have a method in the enum that returns the IDataAccess. like
```
public enum Factory{
    INSTANCE

    public IDataAccess getDataAccess()
    {
        //create and get the data access here.
    }
```